Holovibes/cuda_sources/otsu.cu

```diff
@@ -7,19 +7,19 @@ using uint = unsigned int;
 
 #define NUM_BINS 256
 
-__global__ void histogram_kernel(float* image, int* hist, int imgSize)
+__global__ void histogram_kernel(float* image, uint* hist, int imgSize)
 {
     int idx = blockIdx.x * blockDim.x + threadIdx.x;
     if (idx < imgSize)
-        atomicAdd(&hist[(unsigned char)(image[idx] * NUM_BINS)], 1);
+        atomicAdd(&hist[static_cast<int>(roundf(image[idx] * 256))], 1);
 }
 
 __global__ void _normalise(float* d_input, float min, float max, int size)
 {
     int tid = blockIdx.x * blockDim.x + threadIdx.x;
 
     if (tid < size)
-        d_input[tid] = (int) (((d_input[tid] - min) / (max - min)) * NUM_BINS);
+        d_input[tid] = roundf(d_input[tid] * 255);
```

Comment:
This function is never used and do not renormalize anymore in range [min, max] but in range [0, 255] (assuming d_input value are in range [0,1]) ?